compiler/src/dmd/dinterpret.d

```diff
@@ -1647,8 +1647,16 @@ public:
         if (exceptionOrCant(e))
             return;
 
-        assert(e.op == EXP.classReference);
-        result = ctfeEmplaceExp!ThrownExceptionExp(loc, e.isClassReferenceExp());
+        if (e.op == EXP.classReference)
+        {
+            result = ctfeEmplaceExp!ThrownExceptionExp(loc, e.isClassReferenceExp());
+        }
+        else
+        {
+            exp.error("to `throw` an expression it must be a non-null `class` reference");
+            exp.errorSupplemental("`%s` has type `%s`", exp.toChars(), exp.type.toChars());
```

Comment:
Can you only display this when it's not actually a class type?

Reply:
The compiler eagerly ctfe-s some things so error is useful even when the type is a class.

The original bug report would print `null` instead of `ret()`